provider/tencentcloud/tencentcloud_discover_test.go

```diff
@@ -0,0 +1,37 @@
+package tencentcloud_test
+
+import (
+	"log"
+	"os"
+	"testing"
+
+	discover "github.com/hashicorp/go-discover"
+	"github.com/hashicorp/go-discover/provider/tencentcloud"
+	"github.com/likexian/gokit/assert"
+)
+
+var _ discover.Provider = (*tencentcloud.Provider)(nil)
+var _ discover.ProviderWithUserAgent = (*tencentcloud.Provider)(nil)
+
+func TestAddrs(t *testing.T) {
+	args := discover.Config{
+		"provider":          "tencentcloud",
+		"access_key_id":     os.Getenv("TENCENTCLOUD_SECRET_ID"),
+		"access_key_secret": os.Getenv("TENCENTCLOUD_SECRET_KEY"),
+		"region":            os.Getenv("TENCENTCLOUD_REGION"),
+		"tag_key":           "consul",
+		"tag_value":         "test",
+		"address_type":      "private_v4",
+	}
+
+	if args["access_key_id"] == "" || args["access_key_secret"] == "" || args["region"] == "" {
+		t.Skip("TencentCloud credentials or region info missing")
+	}
+
+	p := &tencentcloud.Provider{}
+	l := log.New(os.Stderr, "", log.LstdFlags)
+
+	addrs, err := p.Addrs(args, l)
+	assert.Nil(t, err)
```

Comment:
We already depend on https://godoc.org/github.com/stretchr/testify and use `assert` and `require` packages in most of our projects.

If we change this to use that it's more consistent with other HC code and doesn't introduce a new dependency into all of our binaries that import this package.

While this is valid for testify/assert and so would only need the import swapped to pass, I suggest changing to use the NoError method which has much more useful output on a failure, and to use `require` instead which will halt the test if it is `nil` rather than possible panic due to continuing and using an undefined return value after an error.

```suggestion
	require.NoError(t, err)
```

The import should change to `github.com/stretchr/testify/require`. I'd personally also change the line below to use `require` rather than having both `assert` and `require` in one test where there is no benefit.